product-service/handlers/get-products-by-id.ts

```diff
@@ -1,36 +1,43 @@
 import { APIGatewayProxyHandler } from 'aws-lambda';
 import 'source-map-support/register';
-import products from '../db/furniture.json';
-
-const headers = {
-  'Access-Control-Allow-Origin': '*',
-  'Access-Control-Allow-Credentials': true,
-}
+import { Client } from 'pg';
+import headers from './headers';
+import generateDbConfig from '../db/db-config';
+import { selectProductById } from '../db/queries';
 
 const getProductsById: APIGatewayProxyHandler = async (event, _context) => {
-  try {
-    console.log('Invoke getProductsById lambda\n');
-    console.log('ENVIRONMENT VARIABLES:' + JSON.stringify(process.env, null, 2));
-    console.info('EVENT:' + JSON.stringify(event, null, 2));
+  console.log('Invoke getProductsById lambda\n');
+  console.log('ENVIRONMENT VARIABLES:' + JSON.stringify(process.env, null, 2));
+  console.info('EVENT:' + JSON.stringify(event, null, 2));
+
+  const dbClient = new Client(generateDbConfig(process));
 
+  try {
+    await dbClient.connect()
     const { productId } = event.pathParameters;
-    const product = products.find(({ id }) => (id === productId));
-  
-    if (!product) {
-      throw new Error(`Product with id: ${productId} not found`);
-    } 
+    const product = await dbClient.query(selectProductById(productId));
```

Comment:
Request https://onka6rul42.execute-api.eu-west-1.amazonaws.com/dev/products/3 causes error 500, which in my opinion is not quite correct. This is because postgresql checks the passed id for correctness and returns an error.

In my opinion it is necessary to check the passed `productId` for compliance with the UUID format and return an error 400 if the passed productId does not match this format.